library/alloc/src/boxed/thin.rs

```diff
@@ -198,7 +223,9 @@ impl<H> WithHeader<H> {
         //    needed to align the header. Subtracting the header size from the aligned data pointer
         //    will always result in an aligned header pointer, it just may not point to the
         //    beginning of the allocation.
-        unsafe { self.0.as_ptr().sub(Self::header_size()) as *mut H }
+        let hp = unsafe { self.0.as_ptr().sub(Self::header_size()) as *mut H };
+        debug_assert!((hp.addr() & (core::mem::align_of::<H>() - 1)) == 0);
+        hp
```

Comment:
What is this checking?

Reply:
That `hp.addr()` is properly aligned.

Reply:
If https://github.com/rust-lang/rust/pull/95643 gets accepted this will be able to use `is_aligned`

Reply:
That’d be great, I’ve wanted that for a long time.

Reply:
It uses that API now.